Sources/Shared/URIFixup.swift

```diff
@@ -59,11 +59,15 @@ public class URIFixup {
   }
 
   public static func getURL(_ entry: String) -> URL? {
-    if let url = URL(string: entry), InternalURL.isValid(url: url) {
-      return URL(string: entry)
+    let trimmed = entry.trimmingCharacters(in: .whitespacesAndNewlines)
+    
+    // NSURL: idnString from brave core handles the puny code represantation of Hostnames
+    // Using Punycode, host names containing Unicode characters are transcoded to a subset of ASCII
+    let entryURL = URL(string: trimmed) ?? NSURL(idnString: trimmed) as URL?
```

Comment:
Do we even want to use `URL(string:)` at all here and just skip to only making the URL via `NSURL(idnString:)`

Reply:
Actually we dont want this because what `NSURL(idnString:)` is doing for IP addresses.

like `http://127.0.1` will be auto converted to `http://127.0.0.1` or

`http://1.1:80` will be "http://1.0.0.1:80"

@Brandon-T correct me If I am wrong here.

Reply:
@soner-yuksel That is correct & valid IP address expansion, makes me think we should definitely use the IDN string initializer by default :)

Reply:
@soner-yuksel Kyle is correct. It should expand. GURL will not make a mistake, so trust it :) we should be banning all usage of URL(string:"..."); soon.
Everywhere should be using the NSURL one.

I will write up an explanation later on WHY with some examples where Foundation.URL is completely busted.

Reply:
Changes at https://github.com/brave/brave-ios/pull/6431/commits/5699c6caa994f515a62a0e09e7ea7bc1a8e7fa9d